### PR TITLE
#14130. Improve call reconnection procedure

### PR DESCRIPTION
**Pull request by alber2510**
- Remove timers from map although the call was destroyed
- Notify to call handler that reconnection process has finished unsuccessful

### CODE REVIEW

**Comment by sergiohs84** [comment]
trigger compilation

File: src/rtcModule/webrtc.cpp [hunk 1]
```diff
@@ -133,13 +133,20 @@ void RtcModule::initInputDevices()
     }
 }
 
-void RtcModule::removeCallRetry(karere::Id chatid)
+void RtcModule::removeCallRetry(karere::Id chatid, bool failed)
```
Comment (sergiohs84) [inline src/rtcModule/webrtc.cpp]: Perhaps it's easier to understand if we refactorize this variable into `retry` (being the opposite value than now).

File: src/rtcModule/webrtc.cpp [hunk 2]
```diff
@@ -1959,18 +1966,25 @@ void Call::destroyIfNoSessionsOrRetries(TermCode reason)
...
-        if (wptr.deleted() || mManager.mRetryCall.find(chatid) == mManager.mRetryCall.end())
+        if (wptrManager.deleted())
+        {
             return;
+        }
 
         mManager.mRetryCall.erase(chatid);
         mManager.mRetryCallTimers.erase(chatid);
 
+        if (wptr.deleted() || mManager.mRetryCall.find(chatid) == mManager.mRetryCall.end())
```
Comment (sergiohs84) [inline src/rtcModule/webrtc.cpp]: The item in the `mRetryCall` has already been erased, so no point to check its existence
```suggestion
        if (wptr.deleted())
```

File: src/rtcModule/webrtc.cpp (hunk 1 above)
Comment (alber2510) [reply src/rtcModule/webrtc.cpp]: Fixed it

File: src/rtcModule/webrtcPrivate.h [hunk 3]
```diff
@@ -385,7 +385,8 @@ class RtcModule: public IRtcModule, public chatd::IRtcHandler
     //=== Implementation methods
     void initInputDevices();
 
-    void removeCallRetry(karere::Id chatid);
+    void updateConstraints(Resolution resolution);
```
[the remark this answers is not in the document]
Comment (alber2510) [reply src/rtcModule/webrtcPrivate.h]: Fixed it

**Comment by alber2510** [comment]
trigger compilation

**Pull request merged**